### ISSUE TITLE
alb ingress controller doesn't work in standard deployment

### ISSUE THREAD

**Issue by autarchprinceps** [aws/eks-charts#284]
When creating just a very simple ingress, the ingress controller in version 1.1.0 (latest released in the helm repo) just fails.
Apparently it cannot connect to itself (dial tcp 192.168.120.75:9443: connect: connection refused). 192.168.120.75 is the pods own IP. Going to newer versions from the git, it never even successfully finishes Helm.

```json
{
"level":"error",
"ts":1602758254.0644438,
"logger":"controller-runtime.controller",
"msg":"Reconciler error",
"controller":"ingress",
"name":"ingress",
"namespace":"monitoring",
"error":"Internal error occurred: failed calling webhook \"mtargetgroupbinding.elbv2.k8s.aws\": Post https://aws-load-balancer-webhook-service.kube-system.svc:443/mutate-elbv2-k8s-aws-v1alpha1-targetgroupbinding?timeout=10s: dial tcp 192.168.120.75:9443: connect: connection refused","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:235\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:209\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1/pkg/internal/controller/controller.go:188\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.18.4/pkg/util/wait/wait.go:90"
}
```

**Comment by M00nF1sh** [comment]
@autarchprinceps 
Did you installed the aws-load-balancer controller chart?

**Comment by autarchprinceps** [comment]
> @autarchprinceps 
> 
> Did you installed the aws-load-balancer-controller chart?

Yes. As I said chart version 1.1.0 from the helm repo for the problem. 1.5.0 from git directly has other issues that prevent it from running at all.

**Comment by kishorj** [comment]
Please install the new CRDs as well before installing 0.1.5 from git directly. What issues do you see with running the latest version?

**Comment by autarchprinceps** [comment]
The container wants to use root, but the chart has explicitly forbidden it. Even when allowing it to, the container doesn't understand the args its given. We have not modified what container it uses. We are using the newest CRDs, though I do not understand, why you have gone back to having to apply CRDs manually. Normally charts have helm do this itself.

**Comment by kishorj** [comment]
> The container wants to use root, but the chart has explicitly forbidden it. Even when allowing it to, the container doesn't understand the args its given. We have not modified what container it uses. We are using the newest CRDs, though I do not understand, why you have gone back to having to apply CRDs manually. Normally charts have helm do this itself.

There have been recent changes to the container permissions. The rc3 image should not require root access. Also, there have been changes to the webhook. The container image and manifest from the helm chart go together. Since there are changes to the manifest across chart versions, the container image used in one chart does not work in the other chart. The chart should only be upgraded (or deleted and installed) via helm commands. If you have a different workflow, please share it across.

Our charts are currently helm v2, so CRD installation is a separate step. Once we have v3 charts, CRDs will be included as  part of helm (there are some limitations though).

Until new charts are released, you can upgrade using the git checkout of rc3 using a similar command to the following -

If using iamserviceaccount
```
helm upgrade -i aws-load-balancer-controller . -n kube-system --set clusterName=<Your Cluster Name> --set serviceAccount.create=false --set serviceAccount.name=aws-load-balancer-controller
```
If not using iamserviceaccount
```
helm upgrade -i aws-load-balancer-controller . -n kube-system --set clusterName=<Your Cluster Name>
```

**Comment by autarchprinceps** [comment]
> The chart should only be upgraded (or deleted and installed) via helm commands. If you have a different workflow, please share it across.

We deploy with Helm Operator via GitOps, so manual steps in between aren't really an option beyond debugging.

> Our charts are currently helm v2, so CRD installation is a separate step. Once we have v3 charts, CRDs will be included as part of helm (there are some limitations though).
Helm v2 is long phased out in all our clusters.

As for the image. I tried again with the version in Git. There is no tag v2.0.0-rc3. After trying various versions, I got it to at least start with rcx7. It manages to create an ALB, but doesn't connect any nodes to it.
service type must be either ‘NodePort’ or ‘LoadBalancer’
The ALB Ingress Controller v1 works with ClusterIP, why isn't this supported anymore in v2?
